src/Support/SupportCliCommand.php

```diff
@@ -0,0 +1,178 @@
+	 *
+	 * [--term_ids=<term_ids>]
+	 * : List of term for which support data need to send. Use comma separator for multiple term ids.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp amp support send-diagnostic
+	 *
+	 * @subcommand send-diagnostic
+	 *
+	 * @param array $args       Positional args.
+	 * @param array $assoc_args Associative args.
+	 */
+	public function send_diagnostic( /** @noinspection PhpUnusedParameterInspection */ $args, $assoc_args ) {
+
+		$is_print     = filter_var( get_flag_value( $assoc_args, 'print', false ), FILTER_SANITIZE_STRING );
+		$is_synthetic = filter_var( get_flag_value( $assoc_args, 'is-synthetic', false ), FILTER_SANITIZE_STRING );
+		$endpoint     = filter_var( get_flag_value( $assoc_args, 'endpoint', '' ), FILTER_SANITIZE_STRING );
+		$endpoint     = untrailingslashit( $endpoint );
+
+		$urls     = filter_var( get_flag_value( $assoc_args, 'urls', false ), FILTER_SANITIZE_STRING );
+		$post_ids = filter_var( get_flag_value( $assoc_args, 'post_ids', false ), FILTER_SANITIZE_STRING );
+		$term_ids = filter_var( get_flag_value( $assoc_args, 'term_ids', false ), FILTER_SANITIZE_STRING );
+
+		$args = [
+			'urls'         => ( ! empty( $urls ) ) ? explode( ',', $urls ) : [],
+			'post_ids'     => ( ! empty( $post_ids ) ) ? explode( ',', $post_ids ) : [],
+			'term_ids'     => ( ! empty( $term_ids ) ) ? explode( ',', $term_ids ) : [],
+			'endpoint'     => $endpoint,
+			'is_synthetic' => $is_synthetic,
+		];
+
+		$this->support_data->set_args( $args );
+		$data = $this->support_data->get_data();
+
+		if ( $is_print ) {
+
+			// Print the data.
+			$print = strtolower( trim( $is_print ) );
+			if ( 'json' === $print ) {
+				echo wp_json_encode( $data ) . PHP_EOL;
+			} else {
+				echo wp_json_encode( $data, JSON_PRETTY_PRINT ) . PHP_EOL;
+			}
+		} else {
+
+			$response = $this->support_data->send_data();
+
+			if ( is_wp_error( $response ) ) {
+				$error_message = $response->get_error_message();
+				WP_CLI::warning( "Something went wrong: $error_message" );
+			} elseif ( empty( $response['status'] ) || 'ok' !== $response['status'] ) {
+				WP_CLI::warning( 'Failed to send diagnostic data.' );
+			} elseif ( ! empty( $response['status'] ) && 'ok' === $response['status'] ) {
+				WP_CLI::success( 'UUID : ' . $response['data']['uuid'] );
+			}
+		}
+
+		/**
+		 * Summary of data.
+		 */
+		$url_error_relationship = [];
+
+		foreach ( $data['urls'] as $url ) {
+			foreach ( $url['errors'] as $error ) {
+				foreach ( $error['sources'] as $source ) {
+					$url_error_relationship[] = $url['url'] . '-' . $error['error_slug'] . '-' . $source;
+				}
+			}
+		}
+
+		$plugin_count = count( $data['plugins'] );
+
+		if ( $is_synthetic ) {
+			$plugin_count_text = ( $plugin_count - 3 ) . " - Excluding common plugins of synthetic sites. ( $plugin_count - 3 )";
+		} else {
+			$plugin_count_text = $plugin_count;
+		}
+
+		$summary = [
+			'Site URL'               => SupportData::get_home_url(),
```

Comment:
Shouldn't this rather be:

```suggestion
			'Site URL'               => $this->support_data->get_home_url(),
```

Reply:
This is intentional since `SupportData::get_home_url()` is static method.

Reply:
Right. The only reason to not do so is that then the dependency injection can't be utilized. In theory, `support_data` may not be a `SupportData` instance. 

Not a big deal, however.